UI/window-basic-main.cpp

```diff
@@ -2014,13 +2014,25 @@ void OBSBasic::InitOBSCallbacks()
 	signalHandlers.emplace_back(
 		obs_get_signal_handler(), "source_filter_add",
 		[](void *data, calldata_t *) {
-			static_cast<OBSBasic *>(data)->UpdateEditMenu();
+			obs_queue_task(
+				OBS_TASK_UI,
+				[](void *param) {
+					static_cast<OBSBasic *>(param)
+						->UpdateEditMenu();
+				},
+				data, false);
```

Comment:
Can you not use `QMetaObject::invokeMethod` with `Qt::QueuedConnection` here like we do in other parts of the UI?

Reply:
that is an option, in that case we need to make `UpdateEditMenu` a slot or `Q_INVOKABLE`